tests/encryption/test_intransit_encryption_sanity.py

```diff
@@ -0,0 +1,57 @@
+import logging
+import pytest
+
+from ocs_ci.ocs.resources.storage_cluster import (
+    in_transit_encryption_verification,
+    set_in_transit_encryption,
+)
+from ocs_ci.framework.pytest_customization.marks import (
+    skipif_intransit_encryption_notset,
+    tier1,
+    skipif_ocs_version,
+)
+
+log = logging.getLogger(__name__)
+
+
+@skipif_intransit_encryption_notset
+class TestInTransitEncryptionSanity:
+    @pytest.fixture(autouse=True)
+    def set_encryption_at_teardown(self, request):
+        def teardown():
+            set_in_transit_encryption()
+
+        request.addfinalizer(teardown)
+
+    @tier1
+    @skipif_ocs_version("<4.12")
+    @pytest.mark.polarion_id("OCS-4861")
+    def test_intransit_encryption_enable_disable_statetransition(self):
+        """
+        The test does the following:
+        1. Verify in-transit Encryption is Enable on setup.
```

Comment:
Why we assume that in transit encryption will be enabled? I need to understand the test flow. Is it enabled at deployment stage? If yes maybe we need a mark to run it only when it is configured in deployment.
If not, let's check if it is enabled and if not enabled it.

Reply:
This probably relates to the question  why we enable it in teardown.

Reply:
All in-transit tests will be run using the deployment configuration [upi_1az_rhcos_vsan_3m_3w_intransit_encryption.yaml](https://github.com/red-hat-storage/ocs-ci/blob/master/conf/deployment/vsphere/upi_1az_rhcos_vsan_3m_3w_intransit_encryption.yaml). Therefore, I am assuming that intransit encryption is already enabled on the setup. I have added the marker skipif_intransit_encryption_notset so that the initial check can be skipped if intransit encryption is not set.

Reply:
Do we have a test that tests the in transit configuration without deployment?
I we have that's great and if we don't we can remove the mark and the flow test could be that we check at start if it enabled or not. If it is enabled then the same flow, if it is not enabled then enabled it and then the rest of the flow - that just in case we don't have cover for none deployment configuration. wdyt?

Reply:
Maybe related to the scenario I've described - if it was i deployment config fail it if we don't have in-transit encryption.

Reply:
I have added the changes as you suggested.